### PR TITLE
Refine prefetch

### PR DESCRIPTION
**Pull request by jacquesqiao**
project: https://github.com/PaddlePaddle/Paddle/issues/10868

### CODE REVIEW

File: paddle/fluid/operators/detail/grpc_server.cc [hunk 1]
```diff
@@ -155,20 +155,22 @@ class RequestPrefetch final : public RequestBase {
 
   void Process() override {
     // prefetch process...
-    std::string varname = request_->OutVarname();
-    VLOG(3) << "RequestPrefetch " << varname;
+    std::string in_var_name = request_->Varname();
+    std::string out_var_name = request_->OutVarname();
+    VLOG(3) << "in_var_name: " << in_var_name
+            << " RequestPrefetch: " << out_var_name;
```
Comment (typhoonzero) [inline paddle/fluid/operators/detail/grpc_server.cc]: Log format should like: `"RequestPrefetch, in_var_name: " << in_var_name << " out_var_name: " << out_var_name`

Comment (jacquesqiao) [reply paddle/fluid/operators/detail/grpc_server.cc]: done

**Comment by typhoonzero** [comment]
Need to fix transpiler tests before merging.

File: paddle/fluid/operators/listen_and_serv_op.cc [hunk 2]
```diff
@@ -135,16 +138,17 @@ void ListenAndServOp::RunSyncLoop(framework::Executor *executor,
...
-      if (blkid != static_cast<size_t>(prefetch_block->ID())) {
-        if (program->Block(blkid).Parent() != last_parent_blkid) {
-          ParallelExecuteBlocks(parallel_blkids, executor, optimize_prepared,
-                                program, recv_scope);
-          parallel_blkids.clear();
-          last_parent_blkid = program->Block(blkid).Parent();
-        }
-        parallel_blkids.push_back(blkid);
+    for (size_t i = 1; i < optimize_block_id_list.size(); ++i) {
+      // skip the first optimize block because it is already in the
```
Comment (Yancey1989) [inline paddle/fluid/operators/listen_and_serv_op.cc]: The comment here is `skip the first optimize block...`, but still start from `i = 1`?

Comment (jacquesqiao) [reply paddle/fluid/operators/listen_and_serv_op.cc]: i is the index of optimize_block_id_list, the optimize_block_id_list[0] is 1, so optimize_block_id_list[1] will not be 1.

Comment (Yancey1989) [reply paddle/fluid/operators/listen_and_serv_op.cc]: Thanks, got it.

File: paddle/fluid/operators/listen_and_serv_op.cc [hunk 3]
```diff
@@ -135,16 +138,17 @@ void ListenAndServOp::RunSyncLoop(framework::Executor *executor,
...
-    for (size_t blkid = 2; blkid < num_blocks; ++blkid) {
-      if (blkid != static_cast<size_t>(prefetch_block->ID())) {
-        if (program->Block(blkid).Parent() != last_parent_blkid) {
-          ParallelExecuteBlocks(parallel_blkids, executor, optimize_prepared,
-                                program, recv_scope);
-          parallel_blkids.clear();
-          last_parent_blkid = program->Block(blkid).Parent();
-        }
-        parallel_blkids.push_back(blkid);
+    for (size_t i = 1; i < optimize_block_id_list.size(); ++i) {
```
Comment (typhoonzero) [inline paddle/fluid/operators/listen_and_serv_op.cc]: maybe should start from 2? block0 is only `listen_and_serv`, and block1 is pushed at line 139

Comment (jacquesqiao) [reply paddle/fluid/operators/listen_and_serv_op.cc]: the first(index=0) item in optimize_block_id_list is 1, the second(index=1) will be 2.

**Pull request merged**